src/lib.rs

```diff
@@ -68,6 +68,7 @@
 //! [`portable-atomic`]: https://crates.io/crates/portable-atomic
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::multiple_bound_locations)] // This is a WONTFIX issue with pin-project-lite
```

Comment:
If this lint is triggered for code generated by external macros, it is definitely clippy bug.

Reply:
Ok, I found https://github.com/rust-lang/rust-clippy/issues/12552.

Reply:
Oh, looking at your PR description, this lint may also need to be allowed on the pin-project-lite side, but it's not clear which one is actually causing a warning here.